### ISSUE TITLE
type 'bool' is not a subtype of type 'Map<dynamic, dynamic>' in type cast

### ISSUE THREAD

**Issue by azol** [koji-1009/flutter_auth_ui#3]
I get an exception on email login: type 'bool' is not a subtype of type 'Map<dynamic, dynamic>' in type cast

s/invokeMapMethod/invokeMethod ?
https://github.com/koji-1009/flutter_auth_ui/blob/98295c220ab41a6134f8d832bf58ec95a32d97ee/lib/flutter_auth_ui.dart#L104

**Comment by koji-1009** [comment]
@azol 
Thanks for reporting!

Sorry, I can not reproduce your crash.
Could you write a snippet of the crash?

I'm assuming code like below.
https://github.com/koji-1009/flutter_auth_ui/blob/master/example/lib/main.dart#L24

**Comment by azol** [comment]
Here you are

```
import 'package:flutter/material.dart';
import 'package:flutter_auth_ui/flutter_auth_ui.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Center(
        child: RaisedButton(
            child: Text('AuthEmail'),
            onPressed: () async {
              print('click');
              final providers = [
                AuthUiItem.AuthEmail,
              ];
              final tosAndPrivacyPolicy = TosAndPrivacyPolicy(
                  "https://www.google.com", "https://www.google.com");

              final result =
                  await FlutterAuthUi.startUi(providers, tosAndPrivacyPolicy);
              print(result);
            }),
      ),
    );
  }
}
```

Log:

```
E/flutter (13350): [ERROR:flutter/lib/ui/ui_dart_state.cc(166)] Unhandled Exception: type 'bool' is not a subtype of type 'Map<dynamic, dynamic>' in type cast
E/flutter (13350): #0      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:161:41)
E/flutter (13350): <asynchronous suspension>
E/flutter (13350): #1      MethodChannel.invokeMethod (package:flutter/src/services/platform_channel.dart:334:12)
E/flutter (13350): #2      MethodChannel.invokeMapMethod (package:flutter/src/services/platform_channel.dart:361:48)
E/flutter (13350): #3      FlutterAuthUi.startUi (package:flutter_auth_ui/flutter_auth_ui.dart:104:33)
E/flutter (13350): #4      MyApp.build.<anonymous closure> (package:firebaseauthexample/main.dart:24:39)
E/flutter (13350): #5      _InkResponseState._handleTap (package:flutter/src/material/ink_well.dart:992:19)
E/flutter (13350): #6      _InkResponseState.build.<anonymous closure> (package:flutter/src/material/ink_well.dart:1098:38)
E/flutter (13350): #7      GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:184:24)
E/flutter (13350): #8      TapGestureRecognizer.handleTapUp (package:flutter/src/gestures/tap.dart:524:11)
E/flutter (13350): #9      BaseTapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:284:5)
E/flutter (13350): #10     BaseTapGestureRecognizer.handlePrimaryPointer (package:flutter/src/gestures/tap.dart:219:7)
E/flutter (13350): #11     PrimaryPointerGestureRecognizer.handleEvent (package:flutter/src/gestures/recognizer.dart:477:9)
E/flutter (13350): #12     PointerRouter._dispatch (package:flutter/src/gestures/pointer_router.dart:78:12)
E/flutter (13350): #13     PointerRouter._dispatchEventToRoutes.<anonymous closure> (package:flutter/src/gestures/pointer_router.dart:124:9)
E/flutter (13350): #14     _LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:377:8)
E/flutter (13350): #15     PointerRouter._dispatchEventToRoutes (package:flutter/src/gestures/pointer_router.dart:122:18)
E/flutter (13350): #16     PointerRouter.route (package:flutter/src/gestures/pointer_router.dart:108:7)
E/flutter (13350): #17     GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:220:19)
E/flutter (13350): #18     GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:200:22)
E/flutter (13350): #19     GestureBinding._handlePointerEvent (package:flutter/src/gestures/binding.dart:158:7)
E/flutter (13350): #20     GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:104:7)
E/flutter (13350): #21     GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:88:7)
E/flutter (13350): #22     _rootRunUnary (dart:async/zone.dart:1206:13)
E/flutter (13350): #23     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
E/flutter (13350): #24     _CustomZone.runUnaryGuarded (dart:async/zone.dart:1005:7)
E/flutter (13350): #25     _invoke1 (dart:ui/hooks.dart:267:10)
E/flutter (13350): #26     _dispatchPointerDataPacket (dart:ui/hooks.dart:176:5)
E/flutter (13350): 
```

I'm on flutter dev

```
Flutter 1.21.0-1.0.pre • channel dev • https://github.com/flutter/flutter.git
Framework • revision f25bd9c55c (7 days ago) • 2020-07-14 20:26:01 -0400
Engine • revision 99c2b3a245
Tools • Dart 2.9.0 (build 2.9.0-21.0.dev 20bf2fcf56)
```

**Comment by koji-1009** [comment]
> I'm on flutter dev

Sorry,  I haven't tested on the dev channel.

**Comment by azol** [comment]
exactly the same on beta
```
Flutter 1.20.0-7.2.pre • channel beta • https://github.com/flutter/flutter.git
Framework • revision a2bde82fbd (4 days ago) • 2020-07-18 15:16:35 -0700
Engine • revision 60b269d898
Tools • Dart 2.9.0 (build 2.9.0-21.2.beta)
```

exactly the same on stable

```
Flutter 1.17.5 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 8af6b2f038 (3 weeks ago) • 2020-06-30 12:53:55 -0700
Engine • revision ee76268252
Tools • Dart 2.8.4
```

**Comment by koji-1009** [comment]
Oh, I misunderstood. Fix this issue.

**Comment by koji-1009** [comment]
@azol 
Hi, I published v `0.2.2`.
https://pub.dev/packages/flutter_auth_ui/versions/0.2.2

Thanks!
